src/components/AssignmentCreatorForm.jsx

```diff
@@ -0,0 +1,141 @@
+  } else if (!date.parsedDate) {
+    errors.date = t('assignment-creator.date-not-valid');
+  } else if (date.parsedDate < Date.now()) {
+    errors.date = t('assignment-creator.past-date-not-valid');
+  }
+  return errors;
+}
+
+function parseDate(value) {
+  const parsedResponse = chrono.parse(value);
+  let parsedDate;
+  if (parsedResponse[0]) {
+    parsedDate = parsedResponse[0].start.date();
+  }
+  return {
+    string: value,
+    parsedDate,
+  };
+}
+
+function formatDate(value) {
+  if (value) {
+    return value.string;
+  }
+  return '';
+}
+
+function AssignmentCreatorForm({
+  courses,
+  handleSubmit,
+  parsedDate,
+  onAssignAssignment,
+  onDraftAssignment,
+  onCloseAssignmentCreator,
+}) {
+  return (
+    <form>
+      <div>
+        <Field
+          component={AssignmentCreatorSelectField}
+          name="course"
+        >
+          <option value="">{t('assignment-creator.select-class')}</option>
+          {
+            courses.map(course => (
+              <option
+                key={course.id}
+                value={course.id}
+              >
+                {course.name}
+              </option>
+            )).valueSeq()
+          }
+        </Field>
+      </div>
+      <div>
+        <Field
+          component={AssignmentCreatorTextField}
+          format={formatDate}
+          name="date"
+          parse={parseDate}
+          parsedDate={parsedDate}
+          placeholder={t('assignment-creator.input-placeholder')}
+          type="text"
+        />
+      </div>
+      <button
+        className={classnames(
+          'assignment-creator__button',
+          'assignment-creator__button_reject',
+        )}
+        type="button"
+        onClick={onCloseAssignmentCreator}
+      >
+        {t('assignment-creator.cancel-button')}
+      </button>
+      <button
+        className={classnames(
+          'assignment-creator__button',
+          'assignment-creator__button_confirm',
+        )}
+        type="button"
+        onClick={handleSubmit(onDraftAssignment)}
+      >
+        {t('assignment-creator.draft-button')}
+      </button>
+      <button
+        className={classnames(
+          'assignment-creator__button',
+          'assignment-creator__button_confirm',
+        )}
+        type="button"
+        onClick={handleSubmit(onAssignAssignment)}
+      >
+        {t('assignment-creator.assign-button')}
+      </button>
+    </form>
+  );
+}
+
+AssignmentCreatorForm.propTypes = {
+  courses: ImmutablePropTypes.iterable.isRequired,
+  handleSubmit: PropTypes.func.isRequired,
+  parsedDate: PropTypes.instanceOf(Date),
+  onAssignAssignment: PropTypes.func.isRequired,
+  onCloseAssignmentCreator: PropTypes.func.isRequired,
+  onDraftAssignment: PropTypes.func.isRequired,
+};
+
+AssignmentCreatorForm.defaultProps = {
+  parsedDate: null,
+};
+
+
+export default reduxForm({
+  form: 'assignmentCreator',
+  validate,
+})(AssignmentCreatorForm);
```

Comment:
Would it make sense to make a `src/containers/AssignmentCreatorForm.js` that did the `reduxForm` work? I don’t see any reason containers specifically have to be using `react-redux`’s `connect()` directly—applying `reduxForm` seems analogous?